File: examples/node/fullstack-incident-kill-angular/peer.mjs
        /**
         * Kill the Path Everywhere — JVM, Agent, React BFF, Angular Admin
         * angular Node peer — same Team profile as Java/Python/other SDKs.
         * Never put Connect tokens in the browser. This process holds identity.
         *
         *   npm install && export KIPONOS_ID=… KIPONOS_ACCESS=… KIPONOS=…
         *   node peer.mjs [value]
         */
        import { Kiponos } from "@kiponos/angular/server";

        const value = process.argv[2] || "on";
        const kip = Kiponos.createFromEnv({ quiet: true });
        await kip.connect();
        await kip.ensurePath("incident");
const path = kip.path("incident");
        await path.set("path-enabled", String(value));
        await path.set("path-enabled-set-at", new Date().toISOString());
        await path.set("path-enabled-set-by", "node-angular-peer");
        await path.set("path-enabled-peer", "angular");

        console.log("same Team leaf incident/path-enabled =", path.get("path-enabled"));
        console.log("  set-by", path.get("path-enabled-set-by"), "teamId", kip.teamId);
        kip.disconnect();
        process.exit(0);

File: examples/python/agentic-med-1016-mi-shopping-pause/peer.py
#!/usr/bin/env python3
"""I Restarted Cursor to Flip Shopping incident pause as a leaf — The Shopping Wall Already Knew

Python peer on the same Team tree as Java / React-Node / Angular-Node.
Hub: examples/agentic-med-1016-mi-shopping-pause/incident-pause (default off)
Product: shopping · Agent host: Cursor
Pain: Shopping freeze waited for an MCP restart

  python3 peer.py                  # local decide()
  python3 -m pytest -q             # logic tests (no tokens)
  KIPONOS_LIVE=1 python3 peer.py   # live hub get/set
"""
from __future__ import annotations

import os
import sys
from pathlib import Path

KEY = "incident-pause"
DEFAULT = "off"
PATH = "examples/agentic-med-1016-mi-shopping-pause/incident-pause"
PEERS = ["java", "python", "react-node", "angular-node"]


def normalize(raw: str | None) -> str:
    v = (DEFAULT if raw is None else str(raw)).strip()
    return v or DEFAULT


def decide(value: str | None) -> dict:
    v = normalize(value)
    paused = v.lower() in ("on", "paused", "yes", "true")
    return {"path": PATH, "key": KEY, "value": v, "action": "freeze_shopping_writes" if paused else "shopping_path_live", "proceed": not paused, "peers": PEERS}


def _live(d: dict) -> None:
    repo = Path(__file__).resolve().parents[3]
    agent = repo / "agent-kit"
    if agent.is_dir() and str(agent) not in sys.path:
        sys.path.insert(0, str(agent))
    from kiponos import Kiponos  # type: ignore

    k = Kiponos.connect(quiet=True)
    try:
        k.ensure_path("examples/agentic-med-1016-mi-shopping-pause")
        live = k.get(PATH, DEFAULT)
        got = decide(str(live) if live is not None else DEFAULT)
        print("live", PATH, "=", got["value"], "action=", got["action"])
    finally:
        k.disconnect()


def main(argv: list[str] | None = None) -> int:
    argv = list(sys.argv[1:] if argv is None else argv)
    raw = argv[0] if argv else None
    d = decide(raw)
    print(f"{d['path']} => {d['value']} action={d['action']} proceed={d['proceed']}")
    print("peers:", ",".join(d["peers"]))
    if os.environ.get("KIPONOS_LIVE") == "1":
        try:
            _live(d)
        except Exception as e:
            print("live optional skipped:", e)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
